src/libraries/System.Private.CoreLib/src/System/IO/Path.Unix.cs

```diff
@@ -58,9 +40,32 @@ public static string GetFullPath(string path, string basePath)
                 throw new ArgumentException(SR.Argument_InvalidPathChars);
 
             if (IsPathFullyQualified(path))
-                return GetFullPath(path);
+                return GetPartiallyQualifiedPath(path, true);
+
+            return GetPartiallyQualifiedPath(CombineInternal(basePath, path), true);
+        }
 
-            return GetFullPath(CombineInternal(basePath, path));
+        internal static string GetPartiallyQualifiedPath(string path, bool checkedForInvalids)
```

Comment:
This can be private.

Reply:
I think a better name for this method would be `GetFullPathInternal`. Also, comment what it does would be nice (GetFullPath without argument validation).